File: app/src/main/java/com/robotnec/budget/app/persistence/schema/CategorySchema.java
package com.robotnec.budget.app.persistence.schema;

import com.yahoo.squidb.annotations.ColumnSpec;
import com.yahoo.squidb.annotations.PrimaryKey;
import com.yahoo.squidb.annotations.TableModelSpec;

@TableModelSpec(className = "CategoryRecord", tableName = "category")
public class CategorySchema {

    @PrimaryKey
    private long id;

    @ColumnSpec(constraints = "NOT NULL UNIQUE")
    private String name;
}

File: app/src/main/java/com/robotnec/budget/core/persistence/schema/CategorySchema.kt
package com.robotnec.budget.core.persistence.schema

import com.yahoo.squidb.annotations.ColumnSpec
import com.yahoo.squidb.annotations.PrimaryKey
import com.yahoo.squidb.annotations.TableModelSpec

@TableModelSpec(className = "CategoryRecord", tableName = "category")
class CategorySchema {

    @PrimaryKey
    private val id: Long = 0

    @ColumnSpec(constraints = "NOT NULL UNIQUE")
    private val name: String? = null
}